src/main/kotlin/com/google/container/tools/diagnostics/GoogleFeedbackErrorReporter.kt

```diff
@@ -0,0 +1,214 @@
+        return ErrorReporterBundle.message("error.googlefeedback.message")
+    }
+
+    override fun submit(
+        events: Array<IdeaLoggingEvent>,
+        additionalInfo: String?,
+        parentComponent: Component,
+        consumer: Consumer<SubmittedReportInfo>
+    ): Boolean {
+        return doSubmit(
+            events[0],
+            parentComponent,
+            consumer,
+            IdeaLogger.ourLastActionId,
+            additionalInfo
+        )
+    }
+
+    companion object {
+
+        @VisibleForTesting const val NONE_STRING = "__NONE___"
+        @VisibleForTesting const val ERROR_MESSAGE_KEY = "error.message"
+        private const val ERROR_STACKTRACE_KEY = "error.stacktrace"
+        @VisibleForTesting const val ERROR_DESCRIPTION_KEY = "error.description"
+        @VisibleForTesting const val LAST_ACTION_KEY = "last.action"
+        private const val OS_NAME_KEY = "os.name"
+        private const val JAVA_VERSION_KEY = "java.version"
+        private const val JAVA_VM_VENDOR_KEY = "java.vm.vendor"
+        @VisibleForTesting const val APP_NAME_KEY = "app.name"
+        @VisibleForTesting const val APP_CODE_KEY = "app.code"
+        @VisibleForTesting const val APP_NAME_VERSION_KEY = "app.name.version"
+        @VisibleForTesting const val APP_EAP_KEY = "app.eap"
+        @VisibleForTesting const val APP_INTERNAL_KEY = "app.internal"
+        @VisibleForTesting const val APP_VERSION_MAJOR_KEY = "app.version.major"
+        @VisibleForTesting const val APP_VERSION_MINOR_KEY = "app.version.minor"
+        private const val PLUGIN_VERSION = "plugin.version"
+
+        private fun doSubmit(
+            event: IdeaLoggingEvent,
+            parentComponent: Component,
+            callback: Consumer<SubmittedReportInfo>,
+            lastActionId: String,
+            description: String?
+        ): Boolean {
+            val intelliJAppNameInfo = ApplicationNamesInfo.getInstance()
+            val intelliJAppExtendedInfo = ApplicationInfoEx.getInstanceEx()
+
+            val params = buildKeyValuesMap(
+                event,
+                description,
+                lastActionId,
+                intelliJAppNameInfo,
+                intelliJAppExtendedInfo,
+                ApplicationManager.getApplication()
+            )
+
+            val dataContext = DataManager.getInstance().getDataContext(parentComponent)
+            val project = CommonDataKeys.PROJECT.getData(dataContext)
+
+            val successCallback: (String) -> Unit = { token ->
+                val reportInfo = SubmittedReportInfo(
+                    null,
+                    "Issue $token",
+                    SubmittedReportInfo.SubmissionStatus.NEW_ISSUE
+                )
+                callback.consume(reportInfo)
+
+                ReportMessages.GROUP
+                    .createNotification(
+                        ReportMessages.ERROR_REPORT, "Submitted", NotificationType.INFORMATION, null
```

Comment:
use `variableName = null` for clarity in this call?

Reply:
I don't think I can since its a non-kotlin function its calling. I can use a comment though